### PR TITLE
Improve ecma_reject to give more detailed error messages

### PR DESCRIPTION
**Pull request by rerobika**
Note: TypedArray.[[DefineOwnProperty]] has been slightly reworked (without semantical changes) for better error messages.

JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

**Review by dbatyai — APPROVED** [review]

File: jerry-core/ecma/operations/ecma-objects.h [hunk 1]
```diff
@@ -27,6 +27,35 @@
...
+
+/**
+ * Reject with TypeError depending on is_throw flags wit the given message
+ */
+#define ECMA_REJECT(is_throw, msg) \
+  ((is_throw) ? ecma_raise_type_error (NULL) : ECMA_VALUE_FALSE)
+#endif /* JERRY_ERROR_MESSAGES */
+
+ecma_value_t ecma_raise_property_redefinition (ecma_string_t *property_name_p, uint16_t flags);
+ecma_value_t ecma_raise_readonly_assignment (ecma_string_t *property_name_p, bool is_throw);
```
Comment (galpeter) [inline jerry-core/ecma/operations/ecma-objects.h:57]: To make it consistent should we also pass `flags` here? (just like in the `ecma_raise_property_redefinition` method)

Comment (rerobika) [reply jerry-core/ecma/operations/ecma-objects.h:57]: Currently `ecma_raise_readonly_assignment ` is only used by `[[Put]]` method where the `is_throw` is passed as a boolean. we should use a ternary to convert it to flags, so I'd leave as it for now.

**Comment by rerobika** [comment]
The patch is rebased.

File: jerry-core/ecma/operations/ecma-objects.h (hunk 1 above)
Comment (galpeter) [reply jerry-core/ecma/operations/ecma-objects.h:57]: True, ok. We'll take a look at it later if we really want to change it.

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**